build/go_check.sh

```diff
@@ -26,26 +26,45 @@ if [[ "${f}" ]]; then
   exit 1
 fi
 
+changes=false
+
 # Check dependencies up-to-date
+# It's okay to remove go.sum, it gets regenerated, and this way it removes unnecessary sums.
 cp go.mod go.mod.prev
+mv go.sum go.sum.prev
 go mod download
-outdated=$(go get -u ./... 2>&1)
+go get -u ./...
+go mod tidy
 go_mod_diff=$(diff -u go.mod go.mod.prev) || true
 if [[ "${go_mod_diff}" ]]; then
+  changes=true
   cat <<EOF
-The following dependencies are out of date:
-${outdated}
 
-There are the following changes:
+go.mod has changes:
 ${go_mod_diff}
 
-Please run 'go get -u ./...' locally and commit the changes.
 EOF
-  exit 1
 fi
 rm go.mod.prev
 
-go mod tidy
+# Check go.sum as well
```

Comment:
why do we need to check go.sum? If we do a go get -u ./... and then go mod tidy, should that clean up go.sum and make it deterministic across envs?

Reply:
+1. Let's just stick to go get + go mod tidy as that is the recommendation per go docs

Reply:
I initially thought that go mod tidy wasn't removing values from go.sum. It seemed like it would just add new checksums, but not remove old ones. I did some more experimenting and it seems like it does, so I'll close this PR.